src/modules/commander/commander.cpp

```diff
@@ -867,12 +867,10 @@ static void commander_set_home_position(orb_advert_t &homePub, home_position_s &
 	mavlink_log_info(mavlink_fd, "home: %.7f, %.7f, %.2f", home.lat, home.lon, (double)home.alt);
 
 	/* announce new home position */
-	if (homePub != nullptr) {
-		orb_publish(ORB_ID(home_position), homePub, &home);
-
-	} else {
-		homePub = orb_advertise(ORB_ID(home_position), &home);
-	}
+    if (homePub == nullptr) {
```

Comment:
Not sure if this was needed or now. Seems like if you have the value, you should publish it right away. not the next time you get it.